qp_calc_period : 0.039
